src/on_call.rs

```diff
@@ -0,0 +1,92 @@
+use curl::easy::*;
+use serde_json;
+use std::env;
+use std::io::prelude::*;
+
+use util::*;
+
+#[derive(Serialize, Debug)]
+#[serde(rename_all = "snake_case", tag = "event_type")]
+pub enum Event {
+    Trigger {
+        incident_key: Option<String>,
+        description: String,
+    },
+    Acknowledge {
+        incident_key: String,
+        description: Option<String>,
+    },
+    Resolve {
+        incident_key: String,
+        description: Option<String>,
+    },
+}
+
+impl Event {
+    /// Sends the event to pagerduty.
+    ///
+    /// If the variant is `Trigger`, this will page whoever is on call
+    /// (potentially waking them up at 3 AM).
+    pub fn send(self) -> CargoResult<()> {
+        let api_token = env::var("PAGERDUTY_API_TOKEN")?;
+        let service_key = env::var("PAGERDUTY_INTEGRATION_KEY")?;
```

Comment:
Ah yeah, I didn't see this in my first skim. I'd like to see this code be a no-op if these variables aren't set, rather than returning an error, so that pagerduty usage is optional.

Reply:
I'm not sure I understand what you mean. This code is only ever going to get called from separate processes used in production to monitor metrics. This code compiles regardless of whether it's configured or not, and would never be called from code paths used outside of our monitors. If that process is run and not configured, we certainly want to be informed immediately rather than have our monitoring silently broken.